presto-kafka/src/main/java/com/facebook/presto/kafka/KafkaSplitManager.java

```diff
@@ -100,7 +99,8 @@ public ConnectorSplitSource getSplits(ConnectorTransactionHandle transaction, Co
                 SimpleConsumer leaderConsumer = consumerManager.getConsumer(partitionLeader);
                 // Kafka contains a reverse list of "end - start" pairs for the splits
 
-                List<HostAddress> partitionNodes = ImmutableList.copyOf(Lists.transform(part.isr(), KafkaSplitManager::brokerToHostAddress));
+                // Always talk to partition leader
+                List<HostAddress> partitionNodes = ImmutableList.of(partitionLeader);
```

Comment:
Ya, lets switch it to a single address, and `getAddresses()` can return `ImmutableList.of(address)`. It's not called in too many places, so the performance shouldn't matter